### PR TITLE
Fix #99

### LINKED ISSUE #99: TypeError when using pointerdown on Sprite
### Description

I'm trying to add typing to a class that uses react-pixi-fiber's Sprite and I get the following error:

```
Type error: Type '{ x: number; y: number; width: number; height: number; texture: Texture; anchor: any; rotation: number; pointerdown: (event: any) => void; interactive: true; buttonMode: true; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Sprite> & Readonly<PropsWithChildren<SpriteProperties>>'.
  Property 'pointerdown' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Sprite> & Readonly<PropsWithChildren<SpriteProperties>>'.  TS2322

    22 |
    23 |     return (
  > 24 |       <Sprite
       |        ^
    25 |         x={node.x} y={node.y}
    26 |         width={NODE_SIZE} height={NODE_SIZE}
    27 |         texture={PIXI.Texture.fromImage(node.track ? node.track.image_cover || DEFAULT_IMAGE_PATH : DEFAULT_IMAGE_PATH)}
```

Has anyone run into the same or a similar typing issue?

### Relevant Code

class with offending error
```
export default class NodeSprite extends React.Component<NodeProps, NodeState> {
  onNodeClick = (event: any) => {
    // implementation
  }

  render() {
    const { node, currentNode } = this.props;

    return (
      <Sprite
        x={node.x} y={node.y}
        width={NODE_SIZE} height={NODE_SIZE}
        texture={PIXI.Texture.fromImage(node.track ? node.track.image_cover || DEFAULT_IMAGE_PATH : DEFAULT_IMAGE_PATH)}
        anchor={centerAnchor}
        pointerdown={this.onNodeClick}
        interactive
        buttonMode
      />
    );
  }
}

```

tsconfig.json

```

```

### Additional info

- react-pixi-fiber version: 0.7.0
- React version: ^16.7.0
- ReactDOM version: ^16.7.0
- PixiJS version: ^4.8.6

### PR DESCRIPTION
**Pull request by eh-steve**
An ugly solution for https://github.com/michalochman/react-pixi-fiber/issues/99, but one which doesn't require changes to pixi.js type definitions

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -84,9 +116,7 @@ declare module 'react-pixi-fiber' {
   export class Sprite extends React.Component<SpriteProperties> {}
 
   /** `Text` component properties */
-  export interface TextProperties extends ChildlessComponent<Omit<PIXI.Text, 'anchor'>> {
-    anchor?: string | number[] | PIXI.ObservablePoint;
-  }
+  export interface TextProperties extends ChildlessComponent<PIXI.Text & InteractiveComponent> {}
```
Comment (michalochman) [inline index.d.ts]: This change removes what was done in #107, any particular reason for that?

File: index.d.ts [hunk 2]
```diff
@@ -21,6 +22,37 @@ declare module 'react-pixi-fiber' {
     children?: React.ReactNode;
   }
 
+  /**
+   * Extra properties to add to allow us to set event handlers using props
+   */
+  export interface InteractiveComponent {
```
Comment (michalochman) [inline index.d.ts]: Is this needed because PIXI defines `InteractionEventTypes` as strings?

File: index.d.ts (hunk 1 above)
Comment (eh-steve) [reply index.d.ts]: Oops, no idea how that happened... I've put it back in in 9f4555b

File: index.d.ts (hunk 2 above)
Comment (eh-steve) [reply index.d.ts]: Yep...

There might be a terser way to accomplish what we need but I wanted to get this working so I could use interactions with TypeScript

**Comment by getkey** [comment]
It looks like  `CustomPIXIComponent` is still affected (but it works fine for the other components exported by react-pixi-fiber).

For the following component:
```ts
// Point.tsx
import { CustomPIXIComponent } from 'react-pixi-fiber';
import { Graphics, interaction } from 'pixi.js';

type Props = {
	fill: number,
	x: number,
	y: number,
	radius: number,
};

export const behavior = {
	customDisplayObject: () => new Graphics(),
		customApplyProps: function(instance: Graphics, oldProps: Props, newProps: Props) {
		const { fill: oldFill, x: oldX, y: oldY, radius: oldRadius, ...remainingOldProps } = oldProps;
		const { fill, x, y, radius, ...remainingNewProps } = newProps;
		instance.clear();

		instance.beginFill(fill);
		instance.drawCircle(x, y, radius);
		instance.endFill();

		// @ts-ignore
		this.applyDisplayObjectProps(remainingOldProps, remainingNewProps);
	},
};
export default CustomPIXIComponent(behavior, 'Point');
```

I get:

```
ERROR in /mnt/data/devel/ble/src/components/InteractivePolygon.tsx
./src/components/InteractivePolygon.tsx
[tsl] ERROR in /mnt/data/devel/ble/src/components/InteractivePolygon.tsx(75,73)
      TS2322: Type '{ fill: number; x: number; y: number; key: number; radius: number; interactive: true; pointerdown: (ev: InteractionEvent) => void; }' is not assignable to type 'IntrinsicAttributes & Props & { children?: ReactNode; }'.
  Property 'interactive' does not exist on type 'IntrinsicAttributes & Props & { children?: ReactNode; }'.
```

**Comment by michalochman** [comment]
Hi @getkey can you please include simplified version of your `InteractivePolygon` component so we can see what are the prop and return types?

**Comment by getkey** [comment]
Sure, here you go

```ts
import React, { FunctionComponent } from 'react';
import { interaction } from 'pixi.js';

import Point from 'src/components/Point';

const InteractivePolygon: FunctionComponent<{}> = () => {
	return (
		<Point
			fill={0x0000ff}
			x={50}
			y={50}
			radius={5}
			interactive
			pointerdown={(ev: interaction.InteractionEvent): void => console.log('hi there', ev)}
		/>
	);
};

export default InteractivePolygon;
```

**Comment by michalochman** [comment]
@getkey can you please check #163? I think your issues now should be fixed. This was a complete rewrite of types so I'd like to make sure I didn't break anything before releasing this.

**Pull request merged**